docs/dev/Debug-LiveTest.md

```diff
@@ -0,0 +1,123 @@
+
+```mermaid
+flowchart TD
+    A[dotnet vstest.console.dll] --> B[dotnet exec AzureMcp.Tests]
+    B --> C[AzureMcp.Tests]
+    C --> D["npm exec azmcp.tz server start"]
+    D --> E["node azmcp server start"]
+```
+
+> **Note:** The actual name of the package `azmcp.tz` may be something like `azure-mcp-0.1.1-alpha.1748757601.tgz` (step 4) is the package containing the `azmcp` server and its command implementations.
+
+To debug the `azmcp` process, it is necessary to:
+
+1. Build the package with debug symbols
+2. Have the `azmcp` (Azure MCP Server) process waits for the Debugger to attach
+3. Ensure the MCP Client does not timeout while waiting to attach to the `azmcp` process
+
+The remainder of this document provides detailed instructions for each of the three steps outlined above.
+
+## Including Debug Symbols
+
+By default, [`/eng/scripts/Build-Module.ps1`](https://github.com/Azure/azure-mcp/blob/a00df9a64d05d274e59d22a689aaf805eef64304/eng/scripts/Build-Module.ps1) builds the `azure-mcp-0.1.1-alpha.xxxxxxxxxx.tgz` package in Release mode. To include debug symbols:
+
+1. Open `Build-Module.ps1`.
+2. Find the [`dotnet publish`](https://github.com/Azure/azure-mcp/blob/a00df9a64d05d274e59d22a689aaf805eef64304/eng/scripts/Build-Module.ps1#L75) command and append `/p:Configuration=Debug`:
+
+```powershell
+$command = "dotnet publish '$projectFile' --runtime '$os-$arch' --output '$outputDir/dist' /p:Version=$Version /p:Configuration=Debug"
+```
+
+## Updating MCP Client Initialization Timeout
+
+1. Open [`LiveTestFixture.cs`](https://github.com/Azure/azure-mcp/blob/a00df9a64d05d274e59d22a689aaf805eef64304/tests/Client/Helpers/LiveTestFixture.cs#L36)
+2. In the `InitializeAsync` method, set a longer timeout:
+
+```csharp
+var clientOptions = new McpClientOptions
+{
+  InitializationTimeout = TimeSpan.FromMinutes(60)
+};
+Client = await McpClientFactory.CreateAsync(clientTransport, clientOptions);
+```
+
+## Updating Azure MCP Server to Wait for Debugger
+
+1. Open [`ServiceStartCommand.cs`](https://github.com/Azure/azure-mcp/blob/a00df9a64d05d274e59d22a689aaf805eef64304/src/Commands/Server/ServiceStartCommand.cs#L40).
+2. Update the `ExecuteAsync` method code to
+   - output the process ID to a file (not standard IO) and,
+   - wait for the debugger to attach.
+
+```csharp
+public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+{
+#if DEBUG
+    LogDebug("[ServiceStartCommand] ExecuteAsync called in debug mode");
+    if (!System.Diagnostics.Debugger.IsAttached)
+    {
+        LogDebug("[ServiceStartCommand] Waiting for debugger to attach. PID: " + Environment.ProcessId);
+        while (!System.Diagnostics.Debugger.IsAttached)
+        {
+            Thread.Sleep(100);
+        }
+        LogDebug("[ServiceStartCommand] Debugger Attached");
+    }
+#else
+    LogDebug("[ServiceStartCommand] ExecuteAsync called in non-debug mode");
+#endif
+...
```

Comment:
Discussed with @joshfree , we can make the debugging experience better by having all these code in azmcp, which get's activated under `#Debug` and `-debug` flag. I'll do that in a followup pr.

Reply:
resolving this, as the code change is now included in the pr.